autopush/router/gcm.py

```diff
@@ -84,7 +85,13 @@ def _route(self, notification, router_data):
                 data['enckey'] = notification.headers.get('encryption-key')
             if ('crypto-key' in notification.headers):
                 data['cryptokey'] = notification.headers.get('crypto-key')
-            data['body'] = notification.data
+            udata = urlsafe_b64encode(notification.data)
```

Comment:
I wonder if we should enforce the encoded length for all routers [in this block of code](https://github.com/mozilla-services/autopush/blob/master/autopush/endpoint.py#L378-L389).

That violates the "Push services MUST NOT return a 413 status code in responses to an entity body that is 4k (4096 bytes) or less in size" provision in the spec...but we'd violate it consistently. :wink: